packages/app-project/src/components/AuthModal/shared/components/withCustomFormik/withCustomFormik.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Formik } from 'formik'
+
+function withCustomFormik (WrappedComponent) {
+  class FormikHOC extends React.Component {
+    constructor () {
+      super()
+
+      this.handleChangeWithCallback = this.handleChangeWithCallback.bind(this)
+    }
+
+    handleChangeWithCallback (event, formikProps) {
+      formikProps.handleChange(event)
+      this.props.onChange(event, formikProps)
+    }
+
+    render () {
+      const { forwardedRef, initialValues, onSubmit, ...rest } = this.props
+      return (
+        <Formik
+          initialValues={initialValues}
+          onSubmit={onSubmit}
+          {...rest}
```

Comment:
```suggestion
          {...optionalProps}
```
Maybe clarify the naming here.